### PR TITLE
Removing "Expect" header from the request

### PR DESCRIPTION
**Pull request by kavishkafernando**
## Purpose
Removing "Expect" header from the request if the request is expecting a 100 continue header.

### CODE REVIEW

File: components/micro-gateway-cli/src/main/resources/templates/service.mustache [hunk 1]
```diff
@@ -72,7 +72,13 @@ service {{cut qualifiedServiceName " "}} on {{#api.transport}}{{#equals . "http"
     }
     @gateway:RateLimit{policy : "{{resourceTier}}"}
     resource function {{operationId}} (http:Caller outboundEp, http:Request req) {
-    gateway:checkExpectHeaderPresent(req);
+    if (req.expects100Continue()) {
+        req.removeHeader("Expect");
+        var result = outboundEp->continue();
+        if (result is error) {
+        log:printError("Error sending response", err = result);
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/service.mustache]: Lets change this error message to add details like error while sending the continue response

File: components/micro-gateway-cli/src/main/resources/templates/service.mustache [hunk 2]
```diff
@@ -72,7 +72,13 @@ service {{cut qualifiedServiceName " "}} on {{#api.transport}}{{#equals . "http"
...
+        }
+    }
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/service.mustache]: Lets move this complete logic to a separate function in each service and generate function name like 
 handleExpectHeaderFor{{cut qualifiedServiceName " "}} and call that function in each resource.

File: components/micro-gateway-cli/src/main/resources/templates/service.mustache (hunk 1 above)
Comment (kavishkafernando) [reply components/micro-gateway-cli/src/main/resources/templates/service.mustache]: fixed

File: components/micro-gateway-cli/src/main/resources/templates/service.mustache (hunk 2 above)
Comment (kavishkafernando) [reply components/micro-gateway-cli/src/main/resources/templates/service.mustache]: fixed

**Pull request merged**